File: backend-nestjs/apps/patient/src/health-metric/health-metric.entity.ts
import { Entity, Column, PrimaryColumn } from 'typeorm';

@Entity('health_metrics')
export class HealthMetricEntity {
  @PrimaryColumn({ type: 'bigint' })
  id: string;

  @Column({ name: 'tenant_id', type: 'bigint' })
  tenantId: string;

  @Column({ name: 'org_id', type: 'bigint' })
  orgId: string;

  @Column({ name: 'patient_id', type: 'bigint' })
  patientId: string;

  @Column({ name: 'metric_type' })
  metricType: string;

  @Column({ name: 'metric_value' })
  metricValue: string;

  @Column({ name: 'recorded_at', type: 'timestamp' })
  recordedAt: Date;
}

File: backend-java/cdm-patient/src/main/java/com/cdm/patient/entity/HealthMetricEntity.java
package com.cdm.patient.entity;

import com.baomidou.mybatisplus.annotation.TableName;
import com.cdm.common.domain.BaseEntity;
import com.cdm.patient.vo.HealthMetricVo;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@TableName("health_metrics")
@Getter
@Setter
@NoArgsConstructor
public class HealthMetricEntity extends BaseEntity {

    private Long tenantId;
    private Long orgId;
    private Long patientId;
    private String metricType;
    private String metricValue;
    private LocalDateTime recordedAt;

    public static HealthMetricVo toVo(HealthMetricEntity entity) {
        if (entity == null) return null;
        return HealthMetricVo.builder()
                .id(entity.getId())
                .tenantId(entity.getTenantId())
                .orgId(entity.getOrgId())
                .patientId(entity.getPatientId())
                .metricType(entity.getMetricType())
                .metricValue(entity.getMetricValue())
                .recordedAt(entity.getRecordedAt())
                .build();
    }
}
